### PR TITLE
feat: build bldr for linux/amd64 and linux/arm64

### PR DESCRIPTION
**Pull request by smira**
`bldr` frontend auto-detects build/target platform and adjusts for it.

Add integration tests for multi-platform builds.

Bump Go, golangci-lint, dependencies, etc. Clean up linting errors.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,13 +4,14 @@ TAG ?= $(shell git describe --tag --always --dirty)
 REGISTRY_AND_USERNAME := $(REGISTRY)/$(USERNAME)
 RUN_TESTS ?= TestIntegration
 
-PLATFORM ?= linux/amd64
+BUILD_PLATFORM ?= linux/amd64
+PLATFORMS ?= linux/amd64,linux/arm64
```
Comment (andrewrynhard) [inline Makefile:8]: super nit: What I tried to do in the template is to have `Makfile` vars match the `buildx` arg. Also other projects have `PLATFORM`. I can live with this but just wanted to point out the reasoning.

Comment (smira) [reply Makefile:8]: makes sense, I will adjust that back to `PLATFORM`!

**Comment by smira** [comment]
```
--- PASS: TestIntegration (84.36s)
    --- PASS: TestIntegration/arch-amd64 (29.82s)
        --- PASS: TestIntegration/arch-amd64/docker-amd64 (8.95s)
        --- PASS: TestIntegration/arch-amd64/llb-amd64 (3.56s)
        --- PASS: TestIntegration/arch-amd64/buildkit-amd64 (17.31s)
    --- PASS: TestIntegration/arch-arm64 (10.32s)
        --- PASS: TestIntegration/arch-arm64/docker-arm64 (1.98s)
        --- PASS: TestIntegration/arch-arm64/llb-arm64 (2.86s)
        --- PASS: TestIntegration/arch-arm64/buildkit-arm64 (5.48s)
    --- PASS: TestIntegration/simple (10.24s)
        --- PASS: TestIntegration/simple/docker (1.99s)
        --- PASS: TestIntegration/simple/buildkit (5.59s)
        --- PASS: TestIntegration/simple/llb (2.65s)
        --- PASS: TestIntegration/simple/validate (0.01s)
    --- PASS: TestIntegration/stages (17.83s)
        --- PASS: TestIntegration/stages/docker-amd64 (2.09s)
        --- PASS: TestIntegration/stages/docker-arm64 (1.99s)
        --- PASS: TestIntegration/stages/docker-multiarch (2.29s)
        --- PASS: TestIntegration/stages/buildkit (5.96s)
        --- PASS: TestIntegration/stages/llb-amd64 (2.77s)
        --- PASS: TestIntegration/stages/llb-arm64 (2.73s)
        --- PASS: TestIntegration/stages/validate (0.00s)
    --- PASS: TestIntegration/variables (16.14s)
        --- PASS: TestIntegration/variables/docker-amd64 (2.02s)
        --- PASS: TestIntegration/variables/docker-arm64 (1.97s)
        --- PASS: TestIntegration/variables/buildkit (6.00s)
        --- PASS: TestIntegration/variables/llb-amd64 (3.20s)
        --- PASS: TestIntegration/variables/llb-arm64 (2.92s)
        --- PASS: TestIntegration/variables/validate (0.02s)
PASS
```

**Comment by smira** [comment]
This now includes support for multi-arch image builds.

For `bldr`-based projects the only change required is bumping frontend version and doing `PLATFORM = linux/amd64,linux/arm64`

**Pull request merged**